lib/snapshot/snapshot_test.go

```diff
@@ -104,3 +104,266 @@ func TestDeleteSnapshotFailed(t *testing.T) {
 		t.Fatalf("Snapshot should have failed, got: %v", err)
 	}
 }
+
+func TestCreateSnapshotWithAuthCredentials(t *testing.T) {
```

Comment:
It seems like this test is mostly testing logic defined in `checkAuthCredentials` function, not the actual authentication validation at `httpserver` package. Wdyt about moving the test to `httpserver` package to check it against a real code?